### DISCUSSION TITLE
Suite names, suite organization and Browserstack

### DISCUSSION THREAD

**Discussion by Mitgenosse** [webdriverio/webdriverio#11912] [Q&A]
Hello there,

We are running webdriverio v8 and have some questions about suite organization and the Browserstack service.

Firstly, we am structuring E2E test as described in the [grouping test specs in suites](https://webdriver.io/docs/organizingsuites/#grouping-test-specs-in-suites) section. E.g. there are test suites like "basicFeatures", "professionalFeatures", etc., which group spec files into a single suite to run sequentially.

The problem is that when running our e2e tests with the flags `--suite basicFeatures --suite professionalFeatures`, it executes all suites one after another, and when an error occurs, the log is not mentioning the name of the suite in which the error occurs.
I couldn't find a way to print the name of the failed suite in the console, so we always have to first find out which suite is problematic. We are using the `spec` and `timeline` reporters.

Secondly, our pipelines execute the E2E tests in Browserstack. We are using the `BROWSERSTACK_BUILD_NAME` which works well in connecting pipeline runs with individual builds in the list on Browserstack.com, like `prod-de pipeline-id:2002728`. The builds in browserstack show different runs in a list, with the name of the latest test spec which ran at the point of finishing (failing or passing) the suite. Is there a way to pass the wdio suite name into it as well, so we directly know which suite we have to check for problems?

<img width="873" alt="Screenshot 2023-12-21 at 16 21 29" src="https://github.com/webdriverio/webdriverio/assets/11715759/677debd5-e2c1-475f-8f9d-a357e355ce0a">

The documentation [on Browserstack-service #sessionNameFormat](https://webdriver.io/docs/browserstack-service/#sessionnameformat) already mentions the default session name of `suiteTitle + ' - ' + testTitle`, but the suite is NOT the suite you define in the "suites" section of the base config file, but the base `describe` block of the `.spec` file. Seems like "suite" has two definitions in the documentation, which is a bit confusing.

Thank you for your time!

**Comment by christian-bromann** [comment]
Hey @Mitgenosse , thanks for raising the discussion. Mind providing a minimal reproducible example we can run and validate on the Browserstack platform?
